### ISSUE TITLE
string function params containing numbers are not quoted

### ISSUE THREAD

**Issue by markwaddle** [StefH/angular-odata-es5#26]
The parameter processing code assumes that strings that can be parsed to numbers should be sent as numeric parameters, ie. without quotes. This seems like a bad assumption, and prevents me from using this code. For example, it prevents the expression of a string param value such as '2017'.

https://github.com/StefH/angular-odata-es5/blob/master/src/angularODataUtils.ts#L23

I propose it instead checks if the type of the value is a string, and if so, wraps it in quotes.

**Comment by StefH** [comment]
This issue is solved by your PR?

**Comment by markwaddle** [comment]
Yes

**Comment by StefH** [comment]
New npm is uploaded which contains both PR.

**Comment by andyward** [comment]
This seems to have regressed for scenarios using numeric entity keys - these now get quoted when using a Get operation. 

Easily demonstrated with this quick karma test, which should fail on 0.24:

``` csharp
    it('Get with numeric key', inject([HttpClient, ODataServiceFactory], (http: HttpClient, factory: ODataServiceFactory) => {
        // Assign
        const service = factory.CreateService<IEmployee>('Employees');

        spyOn(http, 'get').and.returnValue(new Observable<Response>());

        // Act
        const result = service.Get(12).Exec();

        // Assert
        assert.isNotNull(result);
        expect(http.get).toHaveBeenCalledWith(`http://localhost/odata/Employees(12)`, jasmine.any(Object));
        // Actually get `http://localhost/odata/Employees('12')`
    }));
```

As this is a TS library, my fix is to allow `ODataService<T>.Get(...)` to accept a key as `string | number` rather than trying to work out the type downstream in the quoting util. I can put a PR together if you like.

**Comment by StefH** [comment]
Good point.

1] I'll fix this and add tests.
2] Also the `put`, `delete` and `patch` have the same issue.
3] And what about guid or bool ? Please advice.

**Comment by andyward** [comment]
Thanks,

We're not yet using the put/patch actions yet. I just added a couple of tests and verified it does have the issue. 

I've just put a PR in with some tests for the above and fixes. I've tested _string | number | bool and guid_ inputs for GET and PUT

**Comment by StefH** [comment]
New NPM uploaded.

**Comment by andyward** [comment]
Thanks! Can confirm that fixes things my end with 0.25. Thanks for the fast response!
